core/src/main/java/feign/Logger.java

```diff
@@ -164,8 +165,35 @@ protected void log(String configKey, String format, Object... args) {
    */
   public static class JavaLogger extends Logger {
 
-    final java.util.logging.Logger logger =
-        java.util.logging.Logger.getLogger(Logger.class.getName());
+    final java.util.logging.Logger logger;
+
+    /**
+     * @deprecated Use {@link #JavaLogger(String)} instead.
+     *
+     *             This constructor can be used to create just one logger. Example =
+     *             {@code Logger.JavaLogger().appendToFile("logs/first.log")}
+     *
+     *             If you create multiple loggers for multiple clients and provide different files
+     *             to write log - you'll have unexpected behavior - all clients will write same log
+     *             to each file.
+     *
+     *             That's why this constructor will be removed in future.
+     */
+    @Deprecated
+    public JavaLogger() {
+      logger = java.util.logging.Logger.getLogger(Logger.class.getName());
+    }
+
+    /**
+     * Constructor for JavaLogger class
+     * 
+     * @param loggerName - A name for the logger. This should be a dot-separated name and should
+     *        normally be based on the package name or class name of the subsystem, such as java.net
+     *        or javax.swing
+     */
+    public JavaLogger(String loggerName) {
```

Comment:
maybe a constructor that accepts a class instead? like most log frameworks

Reply:
maybe both? I think just a class could be not enough. For example if I have 2 same clients such 
```SomeService client1 = ...; 
SomeService client2 = ...;
```
And in some reason I want to write them to different files. It will force developers to make special class.